metadata-ingestion/src/datahub/ingestion/source/iceberg/iceberg_profiler.py

```diff
@@ -98,60 +103,60 @@ def profile_table(
         Yields:
             Iterator[Iterable[MetadataWorkUnit]]: Workunits related to datasetProfile.
         """
-        if not table.snapshots() or not isinstance(table, BaseTable):
+        if not table.current_snapshot():
             # Table has no data, cannot profile, or we can't get current_snapshot.
             return
 
-        row_count: int = int(table.current_snapshot().summary["total-records"])
-        column_count: int = len(table.schema()._id_to_name)
+        row_count = (
+            int(table.current_snapshot().summary.additional_properties["total-records"])
+            if table.current_snapshot().summary
+            else 0
+        )
+        column_count = len(table.schema()._name_to_id)
```

Comment:
Do you want the total number of fields? Or the number of top-level fields? I would expect the latter. 
```suggestion
        column_count = len(table_schema_nested.fields)
```

Reply:
This is a great question that I've asked before:
<img width="545" alt="image" src="https://user-images.githubusercontent.com/56447460/209816899-7cad4a51-849b-4158-b740-1db7b7325ca1.png">

Reply:
One way is to get all the field IDs and then filter them. We recently added methods to help with this to `schema` so the next release you'd be able to do this:

```python
len([ table.schema.find_field(id).field_type.is_primitive for id in table.schema.field_ids ])
```